controllers/machineset_controller.go

```diff
@@ -382,7 +382,7 @@ func (r *MachineSetReconciler) syncReplicas(ctx context.Context, ms *clusterv1.M
 		// Choose which Machines to delete.
 		machinesToDelete := getMachinesToDeletePrioritized(machines, diff, deletePriorityFunc)
 
-		errCh := make(chan error, diff)
+		errCh := make(chan error, 1)
```

Comment:
Instead of doing this, can we actually fix the code below to read on all the errors and use apimachinery's NewAggregate to aggregate all of them together? 

We can use a for loop on the channel, which we need to close after we're done with it? We might also need to make the channel unbuffered, haven't tested this yet.

Reply:
Addressed your comments.
Please take another look.